### PR TITLE
WIP: Linux mpu9250 rework

### PR DESCRIPTION
**Pull request by nicolaerosia**
Companion to https://github.com/PX4/DriverFramework/pull/212

### CODE REVIEW

**Comment by LorenzMeier** [comment]
Thanks for this major cleanup, it is much appreciated! We certainly want to overall invest into this direction heavily for the next release.

**Comment by TSC21** [comment]
@nicolaerosia can you rebase and fix up the CI issues?

File: .gitmodules [hunk 1]
```diff
@@ -27,7 +27,7 @@
 	url = https://github.com/PX4/Matrix.git
 [submodule "src/lib/DriverFramework"]
 	path = src/lib/DriverFramework
-	url = https://github.com/PX4/DriverFramework.git
+	url = https://github.com/nicolaerosia/DriverFramework.git
```
Comment (TSC21) [inline .gitmodules]: I suppose we need an update on the DriverFramework so we can point it there instead

**Comment by nicolaerosia** [comment]
@TSC21 adding this on my todo. I got this working fine, but I would like to move it from DriverFramework to Firmware to avoid the wrapper.

**Comment by TSC21** [comment]
I think the idea is to centralize all the POSIX drivers on the DriverFramework and not the other way around.

**Comment by nicolaerosia** [comment]
dagar tries to make them portable, I see no value in keeping them in DF ( https://github.com/PX4/Firmware/pull/7864 )

**Comment by TSC21** [comment]
> dagar tries to make them portable, I see no value in keeping them in DF ( #7864 )

I suppose you are right. @LorenzMeier @dagar anything to add here?

**Comment by nicolaerosia** [comment]
we can close this - I'll make a PR

**Comment by dagar** [comment]
The missing piece is the Linux SPI Device interface, which is really just a thin read/write wrapper over spidev. That can be the next step for #7864, and the mpu9250 and ms5611 would be good candidates for the first NuttX SPI drivers to test on Linux.

**Pull request closed without merging**